src/preprocessing/passes/miplib_trick.cpp

```diff
@@ -0,0 +1,657 @@
+            {
+              if ((k & (k - 1)) != 0)
+              {
+                Rational sum = 0;
+                Debug("miplib") << k << " => " << checks[pos_var][k] << endl;
+                for (size_t v = 1, kk = k; kk != 0; ++v, kk >>= 1)
+                {
+                  if ((kk & 0x1) == 1)
+                  {
+                    Assert(pos.getKind() == kind::AND);
+                    Debug("miplib") << "var " << v << " : " << pos[v - 1]
+                                    << " coef:" << coef[pos_var][v - 1] << endl;
+                    sum += coef[pos_var][v - 1];
+                  }
+                }
+                Debug("miplib") << "checkSum is " << sum << " input says "
+                                << checks[pos_var][k] << endl;
+                if (sum != checks[pos_var][k])
+                {
+                  eligible = false;
+                  Debug("miplib") << "  -- INELIGIBLE " << pos
+                                  << " -- (nonlinear combination)" << endl;
+                  break;
+                }
+              }
+              else
+              {
+                Assert(checks[pos_var][k] == 0,
+                       "checks[(%s,%s)][%u] should be 0, but it's %s",
+                       pos.toString().c_str(),
+                       var.toString().c_str(),
+                       k,
+                       checks[pos_var][k]
+                           .toString()
+                           .c_str());  // we never set for single-positive-var
+              }
+            }
+          }
+          if (!eligible)
+          {
+            eligible = true;  // next is still eligible
+            continue;
+          }
+
+          Debug("miplib") << "  -- ELIGIBLE " << v << " , " << pos << " --"
+                          << endl;
+          vector<Node> newVars;
+          expr::NodeSelfIterator ii, iiend;
+          if (pos.getKind() == kind::AND)
+          {
+            ii = pos.begin();
+            iiend = pos.end();
+          }
+          else
+          {
+            ii = expr::NodeSelfIterator::self(pos);
+            iiend = expr::NodeSelfIterator::selfEnd(pos);
+          }
+          for (; ii != iiend; ++ii)
+          {
+            Node& varRef = intVars[*ii];
+            if (varRef.isNull())
+            {
+              stringstream ss;
+              ss << "mipvar_" << *ii;
+              Node newVar = nm->mkSkolem(
+                  ss.str(),
+                  nm->integerType(),
+                  "a variable introduced due to scrubbing a miplib encoding",
+                  NodeManager::SKOLEM_EXACT_NAME);
+              Node geq = Rewriter::rewrite(nm->mkNode(kind::GEQ, newVar, zero));
+              Node leq = Rewriter::rewrite(nm->mkNode(kind::LEQ, newVar, one));
+
+              Node n = Rewriter::rewrite(geq.andNode(leq));
+              assertionsToPreprocess->push_back(n);
+              ProofManager::currentPM()->addDependence(n, Node::null());
```

Comment:
This needs to be protected by an `if (PROOF_ON())` or in a `PROOF()` macro. If CVC4 is compiled without proof support [`currentPM()` will return a `nullptr`](https://github.com/CVC4/CVC4/blob/047e75b485ad16a729083c210ba4064943d2e7c5/src/smt/smt_engine_scope.cpp#L44).

Reply:
Done